File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        Map = dict()

        for word in words:
            for cha in word:
                Map[cha] = Map.get(cha, 0) + 1

        for v in Map.values():
            if v % len(words) != 0:
                return False

        return True

File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.go
func makeEqual(words []string) bool {
	freq := make(map[byte]int, 0)

	for _, word := range words {
		for i := 0; i < utf8.RuneCountInString(word); i++ {
			char := word[i]
			freq[char] += 1
		}
	}

	for _, val := range freq {
		if val%len(words) != 0 {
			return false
		}
	}

	return true
}
